src/server/JasmineGraphServer.cpp

```diff
@@ -597,102 +632,92 @@ void JasmineGraphServer::resolveOperationalGraphs() {
         bcopy((char *)server->h_addr, (char *)&serv_addr.sin_addr.s_addr, server->h_length);
         serv_addr.sin_port = htons(workerPort);
         if (Utils::connect_wrapper(sockfd, (struct sockaddr *)&serv_addr, sizeof(serv_addr)) < 0) {
-            std::cerr << "ERROR connecting" << std::endl;
-            // TODO::exit
             continue;
         }
 
-        bzero(data, FED_DATA_LENGTH + 1);
-        int result_wr = write(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE.c_str(),
-                              JasmineGraphInstanceProtocol::HANDSHAKE.size());
-
-        if (result_wr < 0) {
-            server_logger.log("Error writing to socket", "error");
+        if (!Utils::send_str_wrapper(sockfd, JasmineGraphInstanceProtocol::HANDSHAKE)) {
+            close(sockfd);
+            continue;
         }
+        server_logger.info("Sent: " + JasmineGraphInstanceProtocol::HANDSHAKE);
 
-        server_logger.log("Sent : " + JasmineGraphInstanceProtocol::HANDSHAKE, "info");
-        bzero(data, FED_DATA_LENGTH + 1);
-        read(sockfd, data, FED_DATA_LENGTH);
-        string response = (data);
+        string response = Utils::read_str_trim_wrapper(sockfd, data, FED_DATA_LENGTH);
+        if (response.compare(JasmineGraphInstanceProtocol::HANDSHAKE_OK) != 0) {
+            server_logger.error("Incorrect response. Expected: " + JasmineGraphInstanceProtocol::HANDSHAKE_OK +
```

Comment:
I think its better to have a seperate funtion in `Utils` for log this message (Same text repeats again and again).

```cpp
void logUnexpectedResponse(logger *logger, std::string expected, std::string actual)
```

Reply:
Will be included in next PR

Reply:
Ok